### PR TITLE
Feat/download sample csv

### PR DESCRIPTION
**Pull request by cheongsiuhong**
## Context

This PR contains the page where users can download a sample `.csv` for bulk issuing memos.
This is part of the bulk issue memos workflow.

## Approach

The user is directed to this page after selecting and confirming *Bulk Generate* on the `IssueMemoChooseModePage`.
Clicking `Download CSV` will download a `.csv` files containing the headers required to generate the memos.
The parameters required are pulled from `template.data.paramsRequired`, and the compulsory issuing fields `UIN` and `UIN Type` are added to the headers as well.

**Features**:
User can download the `.csv` with pre-filled headers and fill in the required fields accordingly.

**Future Improvements**:
- Filename can be custom to the template name, but we will have to restrict forbidden characters in the template name beforehand.
- We might want to add a sample row of data like Postman does? Not sure if there's any value in that.
- The colors for this screen are slightly off, but then again the background color for every page is off anyway (it's supposed to be slightly darker).

**Bug Fixes**:

- Some refactoring of constants.
- Minor type fix in `IssueMemoChooseModePage`

## Screenshots

image

**New dependencies**:

- `downloadjs` : Used to download the `.csv` on frontend.

**Bugs**
Commas in the param name will cause the param to become two separate header fields.
image

I will include a hotfix, but this seems like something that should be disallowed during template creation.
Edit: hotfixed by escaping `"` and `,`. The param name is exactly `""keywords", "bug text""`
image

### CODE REVIEW

File: frontend/src/pages/issue/IssueMemoChooseModePage.tsx [hunk 1]
```diff
@@ -38,12 +39,12 @@ export const IssueMemoChooseModePage = (): ReactElement => {
   const handleContinueClick = () => {
     selectedMode === 'single'
       ? navigate(`/issue/${templateId}/single`)
-      : navigate(`/issue/${templateId}/bulk`)
+      : navigate(`/issue/${templateId}/csv`)
```
Comment (dvrylc) [inline frontend/src/pages/issue/IssueMemoChooseModePage.tsx]: Might be a good idea to have all bulk-related routes start with `/bulk`, e.g. `/bulk/download`, `/bulk/upload`

File: frontend/src/pages/issue/IssueMemoSampleCsvPage.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            again in the next page.
+          </Text>
+
+          <Center w="100%" bg="gray.50" p="4" mb="16">
+            <VStack spacing="4">
+              <Icon as={Download} w="12" h="12" />
+              <Text>Download the CSV and fill in the required fields</Text>
+              <Button
+                variant="outline"
+                disabled={!template}
```
Comment (dvrylc) [inline frontend/src/pages/issue/IssueMemoSampleCsvPage.tsx]: If `template` is undefined, `isError` should be true and this wouldn't render at all, so this prop might be unnecessary?

File: frontend/src/pages/issue/IssueMemoSampleCsvPage.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+  const handleDownloadClick = async () => {
+    try {
+      let headers = template?.paramsRequired || []
+      // Params from backend should not have UIN_TOKEN and UIN_TYPE_TOKEN
+      headers = [UIN_TOKEN, UIN_TYPE_TOKEN, ...headers]
+      // Escape " and /
+      const content = headers
+        .map((h) => `"${h.replace(/\\"/g, '""')}"`)
+        .join(',')
```
Comment (dvrylc) [inline frontend/src/pages/issue/IssueMemoSampleCsvPage.tsx]: I might be wrong, but seems like the comment and regex don't match up 🤔 What is the purpose of replacing `\"` with `""`?

File: frontend/src/pages/issue/IssueMemoSampleCsvPage.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+      // TODO: Custom filename? Template name can have illegal characters.
```
Comment (dvrylc) [inline frontend/src/pages/issue/IssueMemoSampleCsvPage.tsx]: I think it's ok to fix the filename as this!

**Review by dvrylc — COMMENTED** [review]
Apologies for the delay. Tested working, left some minor comments!

File: frontend/src/pages/issue/IssueMemoSampleCsvPage.tsx (hunk 3 above)
Comment (cheongsiuhong) [reply frontend/src/pages/issue/IssueMemoSampleCsvPage.tsx]: Added a comment to clarify the escaping, hope it clarifies.

Comment (dvrylc) [reply frontend/src/pages/issue/IssueMemoSampleCsvPage.tsx]: Makes sense, lgtm!

**Review by dvrylc — APPROVED** [review]

**Pull request merged**